### PR TITLE
Use custom i18n utility instead of Angular's `DatePipe`

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
Issue: https://github.com/blackbaud/skyux-i18n/issues/30

### CODE REVIEW

File: src/app/public/modules/date-pipe/date.pipe.ts [hunk 1]
```diff
@@ -19,49 +15,40 @@ import {
...
+  private formattedValue: string;
+
+  private ngUnsubscribe = new Subject<void>();
 
   constructor(
     private changeDetector: ChangeDetectorRef,
     private localeProvider: SkyAppLocaleProvider
   ) {
-    this.locale = this.defaultLocale;
+    this.defaultLocale = 'en-US';
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/date-pipe/date.pipe.ts]: Why is `defaultLocale` set in the constructor, but `defaultFormat` is set up above?

File: src/app/public/modules/date-pipe/date.pipe.spec.ts [hunk 2]
```diff
@@ -84,6 +148,16 @@ describe('Date pipe', () => {
     expect(expectedValues).toContain(value);
   });
 
+  it('should default to mediumDate format', () => {
+    fixture.componentInstance.format = undefined;
+    fixture.detectChanges();
+    const value = fixture.nativeElement.textContent.trim();
+    const expectedValues = [
+      '1/1/2000, 12:00 AM'
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/date-pipe/date.pipe.spec.ts]: Need an IE value here?

File: src/app/public/modules/date-pipe/date.pipe.spec.ts [hunk 3]
```diff
@@ -143,6 +203,20 @@ describe('Date pipe', () => {
...
+    fixture.detectChanges();
+    const value = fixture.nativeElement.textContent.trim();
+    expect(value).toEqual('1/11/2017, 9:25 AM');
+  });
+
+  it('should format invalid in Safari ISO date', () => {
+    fixture.componentInstance.dateValue = '2017-01-20T19:00:00+0000';
+    fixture.detectChanges();
+    const value = fixture.nativeElement.textContent.trim();
+    expect(value).toEqual('1/20/2017, 2:00 PM');
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/date-pipe/date.pipe.spec.ts]: Need an IE check here?

File: src/app/public/modules/date-pipe/date.pipe.spec.ts [hunk 4]
```diff
@@ -143,6 +203,20 @@ describe('Date pipe', () => {
...
     expect(expectedValues).toContain(value);
-    expect(pipe['_locale']).toEqual('en-US');
+    expect(pipe['defaultLocale']).toEqual('en-US');
+  });
+
+  it('should format invalid in IE ISO date', () => {
+    fixture.componentInstance.dateValue = '2017-01-11T09:25:14.014-0500';
+    fixture.detectChanges();
+    const value = fixture.nativeElement.textContent.trim();
+    expect(value).toEqual('1/11/2017, 9:25 AM');
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/date-pipe/date.pipe.spec.ts]: Need an IE check here?

File: src/app/public/modules/date-pipe/date.pipe.spec.ts (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/date-pipe/date.pipe.spec.ts]: Done.

File: src/app/public/modules/date-pipe/date.pipe.spec.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/date-pipe/date.pipe.spec.ts]: Done.

File: src/app/public/modules/date-pipe/date.pipe.spec.ts (hunk 4 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/date-pipe/date.pipe.spec.ts]: Done.

File: src/app/public/modules/date-pipe/date.pipe.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/date-pipe/date.pipe.ts]: Good call. Updated.

**Pull request merged**